ast/compile_test.go

```diff
@@ -109,6 +110,122 @@ func TestCompilerExample(t *testing.T) {
 	assertNotFailed(t, c)
 }
 
+func TestCompilerUserFunctionConflicts(t *testing.T) {
```

Comment:
It's fine if they share the same name, they will resolve separately to `data.package.name` and `package.name`.